### ISSUE TITLE
Feature Request: First Class API Stubs

### ISSUE THREAD

**Issue by davemo** [linemanjs/lineman#318]
I wanted to create this to capture the thoughts while they are fresh in my mind after context on a client project where this would have saved some time:

I think we should formalize a `config/stubs` directory which has the following properties:
1. is watched for changes to files within, triggering a restart of express on change
2. captures syntax errors in watched stub files and catches the exception to avoid killing express
3. (optional) loosely formalize a blessed JSON stub format (simplest thing is `module.exports = {}` for now), we don't need to get too complicated here
4. (optional) auto require stubs based on filename such that they are available within `config/server.{js,coffee}` either as a local in the scope or on the node global object

The first two would be mostly quality of life improvements to the way I use the existing API stubbing feature, the latter two are things that would be really cool if we could make them not suck in implementation ;)

If anyone is willing to pair with me on this, I'd like to start with the first two which are captured in #268.

**Comment by BurningDog** [comment]
> The most annoying part of working in this way is having to restart lineman when a stubbed file changes; changes to anything within drawRoutes seems to be detected with the watchr setup internally in Lineman, but anything that is required within the file that changes is not monitored.

Agreed. I'm not sure how to configure lineman to watch an additional folder for changes, but I edited ```node_modules/lineman/tasks/server.coffee``` and changed the following lines from

```
  resetRoutesOnServerConfigChange = (app) ->
    watchr grunt.file.expand('config/server.*'), (err, watcher) ->
```

to

```
  resetRoutesOnServerConfigChange = (app) ->
    watchr grunt.file.expand('config/server.*', 'config/stubs/**/*'), (err, watcher) ->
```

This allows for file changes without restarting lineman, but for some reason livereload doesn't trigger. I'd love to know how to put this change into ```config/application.js```
